uptane/clients/client.py

```diff
@@ -0,0 +1,413 @@
+      which includes that "custom" field with ECU Serial assignments.
+
+    <Returns>
+      Target file info compliant with tuf.formats.TARGETFILE_INFO_SCHEMA,
+
+
+    <Exceptions>
+
+      tuf.UnknownTargetError
+        if a given filepath is not listed by the consensus of Director and
+        Image Repository (or through whichever trusted path is specified by
+        this client's pinned.json file.) If info is returned, it will match
+        tuf.formats.TARGETFILE_SCHEMA and will have been validated by all
+        required parties.
+
+      tuf.NoWorkingMirrorError
+        will be raised by the updater.target() call here if we are unable to
+        validate reliable target info for the target file specified (if the
+        repositories do not agree, or we could not reach them, or so on).
+
+      uptane.Error
+        if the Director targets file has not provided information about the
+        given target_filepath, but target_filepath has nevertheless been
+        validated. This could happen if the map/pinning file for some reason
+        incorrectly set to not require metadata from the Director.
+
+    """
+    tuf.formats.RELPATH_SCHEMA.check_match(target_filepath)
+
+    validated_target_info = self.updater.target(
+      target_filepath, multi_custom=True)
+
+    # validated_target_info will now look something like this:
+    # {
+    #   'Director': {
+    #     filepath: 'django/django.1.9.3.tgz',
+    #     fileinfo: {hashes: ..., length: ..., custom: {'ecu_serial': 'ECU1010101'} } },
+    #   'ImageRepo': {
+    #     filepath: 'django/django.1.9.3.tgz',
+    #     fileinfo: {hashes: ..., length: ... } } }
+    # }
+
+    # We expect there to be an entry in the dict with key name equal to the
+    # name of the Director repository (specified in pinned.json).
+
+    if self.director_repo_name not in validated_target_info:
+      # TODO: Consider a different exception class. This seems more like an
+      # assert statement, though.... If this happens, something is wrong in
+      # code, or pinned.json is misconfigured (to allow target validation
+      # whereby the Director is not specified in some multi-repository
+      # delegations) or the value of director_repo_name passed to the
+      # initialization of this object was wrong. Those are the edge cases I can
+      # come up with that could cause this.
+
+      # If the Director repo specified as self.director_repo_name is not in
+      # pinned.json at all, we'd have thrown an error during __init__. If the
+      # repos couldn't provide validated target file info, we'd have caught an
+      # error earlier instead.
+
+      raise uptane.Error('Unexpected behavior: did not receive target info from'
+          ' Director repository (' + repr(self.director_repo_name) + ') for '
+          'a target (' + repr(target_filepath) + '). Is pinned.json configured '
+          'to allow some targets to validate without Director approval, or is'
```

Comment:
Please add a space after `is`.

Reply:
On a side-note, it's a good idea to have the separating space between strings over multiple lines either all in the end or in the beginning (I am slightly in favor of the former, as you did in all but one lines here).